src/vs/workbench/api/node/extHostTask.ts

```diff
@@ -195,6 +195,9 @@ namespace TaskDTO {
 			} else {
 				scope = value.scope.uri;
 			}
+		} else {
+			// To continue to support the deprecated task constructor that doesn't take a scope, we must add a scope here:
+			scope = types.TaskScope.Workspace;
 		}
 		if (!definition || !scope) {
```

Comment:
We could remove this || !scope now because it will never be undefined.

Reply:
I'll remove it in an upcoming change in master.